### DISCUSSION TITLE
How to integrate apollo-server to nuxt3? (Graphql)

### DISCUSSION THREAD

**Discussion by noobSam1122** [nuxt/nuxt#16654] [Questions]
Nuxt3 uses Nitro to create server and serve on port 3000, and apollo serve on port 4000, what am I want to do is let apollo listen on 3000 for graphql request, how can I set it up without port conflict?

**Comment by TheAlexLichter** [comment]
You could:
1. set up a proxy that proxies the GraphQL endpoint to port 4000
or
2. Run the GraphQL server as middleware of nitro (if that's possible)

**Comment by Jlaird** [comment]
If you are working on a small project use https://v3.nuxtjs.org/guide/directory-structure/server/ but if you are talking about an enterprise app. I would recommend using a different domain and setting it up as a standalone. This would mean fewer complications when deploying along with better scaling of the services and separating concerns.
